local/src/core/ProxyVerifier.cc

```diff
@@ -2382,6 +2388,10 @@ on_data_chunk_recv_cb(
   }
   H2StreamState &stream_state = *iter->second;
   stream_state._received_body_length += len;
+  if (flags & NGHTTP2_FLAG_END_STREAM) {
+    errata.diag("on_data_chunk_recv_cb End of data for stream {}", stream_id);
+    session_data->set_headers_are_available(stream_id);
+  }
```

Comment:
I noticed this in the docs for nghttp2_on_data_chunk_recv_callback:

>  (flags & NGHTTP2_FLAG_END_STREAM) != 0 does not necessarily mean this chunk of data is the last one in the stream. You should use nghttp2_on_frame_recv_callback to know all data frames are received. 

https://nghttp2.org/documentation/types.html#c.nghttp2_on_data_chunk_recv_callback